Update Relatives subsection so that "I Dont Know" clears branches

christopherhuii opened the pull request:
## Description
Fixes #1466 

If you selected `I Don't Know` for the following questions, all data and branching will be removed.

1. `Provide the address of current employer, or provide the address of their most recent employer if not currently employed`

2. `Is this relative affiliated with a foreign government, military, security, defense industry, foreign movement, or intelligence service?`

## Checklist for Reviewer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

suzubara commented on src/components/Section/Relationships/Relatives/Relative.jsx:
```diff
@@ -19,65 +17,164 @@ import {
+      value: 'Guardian',
+    },
+  ])
+
```
The options defined in these functions seem like they'd be good to store in another file eventually.

suzubara commented:
Looking at the original issue, it looks like the address type (In the U.S.) should be un-selected if you choose "I don't know", however on this branch it still appears to be selected.

christopherhuii commented:
I added the ability for the `Address` component to not have an address type. However, this makes things a bit more complicated in how the address fields display. I added some details [here](https://github.com/18F/e-QIP-prototype/issues/1466#issuecomment-479135538).

I also moved the `Is this relative affiliated with a foreign government, military, security, defense industry, foreign movement, or intelligence service?` explanation texture underneath the `I don't know` button as indicated [here](https://github.com/18F/e-QIP-prototype/issues/1466#issuecomment-478638365).

ryanhofdotgov commented:
@christopherhuii Regarding un-typed Addresses, what changes would the backend see? Is `isEnabled` a new property that's going to start manifesting itself in the serialized JSON?

christopherhuii commented:
`isEnabled` does not get sent to the backend. The backend does not receive anything new in the JSON.

ryanhofdotgov commented:
@christopherhuii @suzubara It seems with this change, `I don't know` has a different behavior in this one section from all the other places where there is an address fieldset/`I don't know` button. For example, Foreign Contacts.

Ditto with other fields (like email) where toggling an associated `I don't know` preserves any existing value in that field. 

What's involved in making the behavior consistent across the app?
